packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.pipeline-with-variables.ts

```diff
@@ -51,8 +51,9 @@ class PipelineStack extends Stack {
 }
 
 const app = new App({
-  context: {
+  postCliContext: {
```

Comment:
I'm not actually aware of the difference between `context` vs `postCliContext`. Can you explain why to use `postCliContext` here?

Reply:
I did this because it didn't work with the actual `context`. I would like to investigate this as well.

Reply:
I've tried putting it back in and some seem to succeed and others fail.

I'll put it back in once.

Reply:
@GavinZZ

Thanks for your merge.
I did a little research and it seems that the following line does not cause a difference even in `context`.

```ts
/// !cdk-integ VariablePipelineStack pragma:set-context:@aws-cdk/core:newStyleStackSynthesis=true
```

https://github.com/go-to-k/aws-cdk/blob/00e8a7b877a732b4236227f6462bf3914ce7a214/packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.pipeline-with-variables.ts#L2

However, using `DefaultStackSynthesizer` in the stack definition seems to cause a difference due to feature flags even if the above line is present.

```ts
const stack = new TestCdkStack(app, 'PipelineSecurityStack', {
  synthesizer: new DefaultStackSynthesizer(),
});
```

https://github.com/go-to-k/aws-cdk/blob/00e8a7b877a732b4236227f6462bf3914ce7a214/packages/@aws-cdk-testing/framework-integ/test/pipelines/test/integ.pipeline-security.ts#L107

I am not sure because I have not read the source code thoroughly...

Regarding "context vs postCliContext", `postCliContext` has priority over context, CLI options, etc.

https://github.com/go-to-k/aws-cdk/blob/00e8a7b877a732b4236227f6462bf3914ce7a214/packages/aws-cdk-lib/core/lib/app.ts#L175

https://github.com/go-to-k/aws-cdk/blob/00e8a7b877a732b4236227f6462bf3914ce7a214/packages/aws-cdk-lib/core/lib/app.ts#L201-L221